pkg/issuer/acme/dns/util/wait.go

```diff
@@ -214,7 +228,10 @@ func ValidateCAA(domain string, issuerID []string, iswildcard bool, nameservers
 					dns.RcodeToString[msg.Rcode], domain)
 			}
 			oldQuery := queryDomain
-			queryDomain = updateDomainWithCName(msg, queryDomain)
+			queryDomain, err := updateDomainWithCName(queryDomain, nameservers)
+			if err != nil {
+				return err
```

Comment:
is it worth annotating the error? e.g.,

```suggestion
				return fmt.Errorf("while trying to follow CNAMEs for domain %s using nameservers %v: %w", queryDomain, nameservers, err)
```